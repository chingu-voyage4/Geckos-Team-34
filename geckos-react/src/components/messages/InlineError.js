import React from 'react';
import { string } from 'prop-types';

const propTypes = {
  message: string.isRequired
};

const InlineError = ({ message }) => {
  return (
    <span style={{ color: '#dc2f2f' }}>{ message }</span>
  );
};

InlineError.propTypes = propTypes;

export default InlineError;