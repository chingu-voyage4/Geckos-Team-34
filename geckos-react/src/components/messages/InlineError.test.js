import React from 'react';
import { mount, shallow } from 'enzyme';

import InlineError from '../messages/InlineError';

describe('<InlineError />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<InlineError message={'This is a test error'} />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.length).toBe(1);
  });

  it('should display a error message', () => {
    const elem = wrapper.find('span');
    expect(elem.length).toBe(1);
    expect(elem.text()).toBe('This is a test error');
  });

});

