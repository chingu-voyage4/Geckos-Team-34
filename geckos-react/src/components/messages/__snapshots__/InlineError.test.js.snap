// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InlineError /> should render correctly 1`] = `
<InlineError
  message="This is a test error"
>
  <span
    style={
      Object {
        "color": "#dc2f2f",
      }
    }
  >
    This is a test error
  </span>
</InlineError>
`;
